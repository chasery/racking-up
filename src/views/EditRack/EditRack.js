import React, { useState } from "react";
import Form from "../../components/Form/Form";
import FormField from "../../components/FormField/FormField";
import "./EditRack.css";

function EditRack(props) {
  const [name, setName] = useState("");

  return (
    <main role="main">
      <section className="EditRack">
        <div className="EditRack__wrapper">
          <Form id="EditRack">
            <h2>Edit Rack</h2>
            <p>Edit the name of the rack.</p>
            <FormField
              id="name"
              label="Name"
              type="text"
              isRequired={true}
              onChange={(e) => setName(e.target.value)}
              value={name}
            />
            <button>Cancel</button>
            <button>Edit Rack</button>
          </Form>
        </div>
      </section>
    </main>
  );
}

export default EditRack;
