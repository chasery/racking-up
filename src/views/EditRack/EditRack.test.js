import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import EditRack from './EditRack';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({
    rackId: 1,
  }),
}));

describe('EditRack component', () => {
  it('renders without error', () => {
    const wrapper = shallow(<EditRack />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
