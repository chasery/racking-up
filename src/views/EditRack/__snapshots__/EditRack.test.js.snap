// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditRack component renders without error 1`] = `
<main
  role="main"
>
  <section
    className="EditRack"
  >
    <div
      className="EditRack__wrapper"
    >
      <Form
        id="EditRack"
      >
        <h2>
          Edit Rack
        </h2>
        <p>
          Edit the name of the rack.
        </p>
        <FormField
          id="name"
          isRequired={true}
          label="Name"
          onChange={[Function]}
          type="text"
          value=""
        />
        <button>
          Cancel
        </button>
        <button>
          Edit Rack
        </button>
      </Form>
    </div>
  </section>
</main>
`;
